### DISCUSSION TITLE
How to handle JSON imports?

### DISCUSSION THREAD

**Discussion by josh-hemphill** [garronej/denoify#54] [General]
I've got an instance where the Tedious package imports its own `package.json` but uses older non-esm target and does not support import assertions.
I'm not sure what the correct way to handle that would be; deno would want import assertions, so rewriting the import to use an assertion might be the right way, but I also wouldn't be apposed to just inlining the package.json text since that would be foolproof.

**Comment by garronej** [comment]
Hi @josh-hemphill,  
Here you have an example of a Node source file that read and parse the package.json: https://github.com/garronej/my_dummy_npm_and_deno_module/blob/master/src/tools/getPackageJsonName.ts  
Denoify is able to sucesfully transpile this to a deno compatible code.  

See here:  
https://github.com/garronej/my_dummy_npm_and_deno_module/blob/master/deno_dist/tools/getPackageJsonName.ts

**Reply by garronej** [reply]
PS: You might want to use [this](https://github.com/garronej/denoify_ci/blob/main/src/tools/getProjectRoot.ts) instead of reading file relative to `__dirname` (once you publish the dir structure might be a bit different than in dev)

**Comment by garronej** [comment]
And when in doubt you always have the option of providing a deno specific implementation for a specific file by creating a `xxx.deno.ts` file.  

See `sha3` [here](https://github.com/garronej/my_dummy_npm_and_deno_module#step-2-enabling-the-modules-hard-dependencies-to-run-on-deno)
